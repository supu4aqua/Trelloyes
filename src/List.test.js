import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

const cards = [{
 id: 1,
  title: 'abc',
  content: 'card-content'
},
{
 id: 2,
  title: 'def',
  content: 'card1-content'
}
]
describe('Card component', () => {
it('renders without crashing', () => {

    const div = document.createElement('div');
    ReactDOM.render(<List cards={cards}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {

  const tree = renderer
    .create(<List cards={cards}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  });
});
