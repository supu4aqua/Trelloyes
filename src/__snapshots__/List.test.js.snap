// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2 />
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        abc
      </h3>
      <p>
        card-content
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        def
      </h3>
      <p>
        card1-content
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;

exports[`renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2 />
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        abc
      </h3>
      <p>
        card-content
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        def
      </h3>
      <p>
        card1-content
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
